esmvaltool/preprocessor/_multimodel.py

```diff
@@ -18,29 +18,22 @@
 from datetime import datetime
 from datetime import timedelta
 import iris
+import cf_units
 import numpy as np
 
 from ._io import save_cubes
 
 logger = logging.getLogger(__name__)
 
 
-def _time_pts_by_calendar(cube):
-    """Extract time points depending on calendar"""
-    # a_p is a prefactor that brings time points
-    # to 365-day calendar
-    time_units = cube.coord('time').units
-    if time_units.calendar is not None:
-        if time_units.calendar == '360_day':
-            a_p = 365. / 360.
-        else:
-            a_p = 1.
-    else:
-        logger.debug("No calendar type: assuming 365-day")
-        a_p = 1.
-    # return tpoints
-    t_points = [c * a_p for c in cube.coord('time').points]
-    return t_points
+def _parse_time_unit(tunit):
+    """Return a datetime object equivalent to tunit"""
+    # tunit e.g. 'day since 1950-01-01 00:00:00.0000000 UTC'
+    # note that we have everything in STANDARD calendar
+    cfunit = cf_units.Unit(tunit,
+                           calendar=cf_units.CALENDAR_STANDARD)
```

Comment:
Why are we assuming standard calendar? Shouldn't we use the calendar attribute in the time coordinate?